#!/usr/bin/env python
PACKAGE = "rqt_quad_control_tuning"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kpxy",    double_t,    0, "PID x/y proportional coefficient", 10.0,  0.0, 30.0)
gen.add("kdxy",    double_t,    0, "PID x/y derivative coefficient", 4.0,  0.0, 30.0)
gen.add("kpz",    double_t,    0, "PID z proportional coefficient", 15.0,  0.0, 30.0)
gen.add("kdz",    double_t,    0, "PID z derivative coefficient", 6.0,  0.0, 30.0)

gen.add("krp",    double_t,    0, "Attitude feedback roll/pitch coefficient", 12.0,  0.0, 30.0)
gen.add("kyaw",    double_t,    0, "Attitude feedback yaw coefficient", 5.0,  0.0, 30.0)

gen.add("rotor_drag",    bool_t,    0, "Perform Aerodynamics Compensation", False)
gen.add("k_drag_xw",    double_t,    0, "Drag coefficient in x/w", 0.0,  0.0, 1.0)
gen.add("k_drag_y",    double_t,    0, "Drag coefficient in y", 0.0,  0.0, 1.0)
gen.add("k_drag_z",    double_t,    0, "Drag coefficient in z", 0.0,  0.0, 1.0)
gen.add("k_thrust_horz",    double_t,    0, "Horizontal thrust coefficient", 0.0, 0.0,  1.0 )

exit(gen.generate(PACKAGE, "rqt_quad_control_tuning", "setControlParameters"))